api/v1alpha1/packagebundle_webhook.go

```diff
@@ -50,10 +55,18 @@ func (r *PackageBundle) ValidateDelete() error {
 }
 
 func (r *PackageBundle) validate() error {
-	valid, digest, yml, err := signature.ValidateSignature(r, signature.EksaDomain)
+	//TODO Turn this into fields in a new CRD used for signature validation configuration
+	keyOverride := os.Getenv(PublicKeyEnvVar)
+	var domain signature.Domain
+	if keyOverride != "" {
+		domain = signature.Domain{Name: signature.DomainName, Pubkey: keyOverride}
+	} else {
+		domain = signature.EksaDomain
+	}
+	valid, digest, yml, err := signature.ValidateSignature(r, domain)
```

Comment:
Your logic here is a little muddied. I'd prefer a more straightforward approach (github won't let me make this as a suggestion, I guess because it touches lines that you didn't modify..?)

```go
valid, digest, yml, err := signature.ValidateSignature(r, domain)
if err != nil {
    return err
}
if !valid {
    return fmt.Errorf("invalid signature for configured public key %q", domain.Pubkey)
}
return nil
```
This deals with one error at a time, and makes it explicit that if you've reached the end of the function, you've succeeded. Alternatively, you _could_ configure `ValidateSignature` to return an error if it fails to validate. 

Additionally, it's generally a bad idea in go to both log and return an error. The top-level caller should log if that makes sense, but it's best to leave that decision to them. If you're going to ignore an error (cuz it's recoverable or whatever), then it can make sense to log that.

Reply:
I normally avoid multiple return points in functions since it tends not to age well. That doesn't seem like a Go idiom though. I've done it in other functions ( I cringe a bit when looking at them ), I could do it here too.

I prefer if `ValidateSignature` returns an error in the case of an actual error and a boolean for validity to avoid trying to match on the error type or message. It allows for the `apilog.Info` which is there to provide details which are not to be returned to the caller and displayed by the `kubectl` output. The log will be available via `kubectl logs` whereas the `error` is printed to `kubectl`'s stdout.

Reply:
> I normally avoid multiple return points in functions since it tends not to age well. That doesn't seem like a Go idiom though. I've done it in other functions ( I cringe a bit when looking at them ), I could do it here too.

You're correct, it's not a go idiom. It's very common in go to return, especially when there's an error. I found it a little off-putting at first, but have since come to appreciate it, as they are generally arranged so as to minimize the amount of info you have to keep in your head at any one time. I believe the egregious uses of return in the middle of a function are for non-error cases. It's a smell that your function is doing more than one thing.

> I prefer if `ValidateSignature` returns an error in the case of an actual error and a boolean for validity to avoid trying to match on the error type or message. It allows for the `apilog.Info` which is there to provide details which are not to be returned to the caller and displayed by the `kubectl` output. The log will be available via `kubectl logs` whereas the `error` is printed to `kubectl`'s stdout.

That all seems well-reasoned. I have some reservations about logging in that way, but it doesn't in any way affect program correctness so 👍🏻 and I'll try to keep my eye on it, and see how it works out.